src/sentry/search/utils.py

```diff
@@ -117,32 +117,41 @@ def parse_datetime_comparison(value):
 
 def parse_datetime_value(value):
     # timezones are not supported and are assumed UTC
-    if value[-1] == 'Z':
+    if value[-1:] == 'Z':
         value = value[:-1]
 
+    result = None
+
     # A value that only specifies the date (without a time component) should be
     # expanded to an interval that spans the entire day.
-    if len(value) in (8, 10):
-        value = datetime.strptime(value, DATE_FORMAT).replace(
-            tzinfo=timezone.utc,
-        )
+    try:
+        result = datetime.strptime(value, DATE_FORMAT).replace(tzinfo=timezone.utc)
+    except ValueError:
+        pass
+    else:
         return (
-            (value, True),
-            (value + timedelta(days=1), False),
+            (result, True),
+            (result + timedelta(days=1), False),
         )
 
     # A value that contains the time should converted to an interval.
-    if value[4] == '-':
+    for format in [DATETIME_FORMAT, DATETIME_FORMAT_MICROSECONDS]:
         try:
-            value = datetime.strptime(value, DATETIME_FORMAT).replace(tzinfo=timezone.utc)
+            result = datetime.strptime(value, format).replace(tzinfo=timezone.utc)
```

Comment:
Don't you want to `break` here or something so as not to always drop into `parse_unix_timestamp` below?

Reply:
Ugh, yes. I had this in an earlier version and still messed it up, even after complaining about for/else the other day